java/client/src/org/openqa/selenium/devtools/target/model/AttachToTarget.java

```diff
@@ -17,44 +17,42 @@
 package org.openqa.selenium.devtools.target.model;
 
 import org.openqa.selenium.json.JsonInput;
+import org.openqa.selenium.remote.SessionId;
 
 import java.util.Objects;
 
 public class AttachToTarget {
 
   private final SessionId sessionId;
 
-  private final TargetId targetId;
+  private final TargetInfo targetInfo;
 
   private final boolean waitForDebugger;
 
   public AttachToTarget(SessionId sessionId,
-                        TargetId targetId, Boolean waitForDebugger) {
+                        TargetInfo targetInfo, Boolean waitForDebugger) {
     this.sessionId = Objects.requireNonNull(sessionId, "sessionId is required");
-    this.targetId = Objects.requireNonNull(targetId, "targetId is required");
+    this.targetInfo = Objects.requireNonNull(targetInfo, "targetInfo is required");
     this.waitForDebugger = Objects.requireNonNull(waitForDebugger, "waitForDebugger is require");
   }
 
   private static AttachToTarget fromJson(JsonInput input) {
-    SessionId sessionId = null;
-    TargetId targetId = null;
+    SessionId sessionId = new SessionId(input.nextString());
```

Comment:
Shouldn't this be inside the loop? And I don't see a call to `input.beginObject`, so I'm a little surprised this works....

Reply:
Also, you added a `fromJson` method, so it should be fine to `input.read(SessionId.class)`

Reply:
this is how the Json Object is build .
I have tested it more the once